pyproject.toml

```diff
@@ -207,3 +218,50 @@ filterwarnings = [
     "ignore:unclosed file <_io.TextIOWrapper name='(/dev/null|nul)' mode='w':ResourceWarning", # OK
     "ignore:Deprecated in traitlets 4.1, use the instance .metadata dictionary directly", # OK (ipywidgets internal)
 ]
+
+[tool.mypy]
+namespace_packages = true
+explicit_package_bases = true
+mypy_path = ""
+exclude = []
```

Comment:
I ran Mypy in the same way you were @MarcSkovMadsen and I got a bunch of additional errors on unrelated files. At least for now, maybe it's worth turning off the unrelated file errors?

```suggestion
exclude = []
follow_imports = "silent"
```

From https://mypy.readthedocs.io/en/stable/running_mypy.html#following-imports:

> If you are starting a new codebase and plan on using type hints from the start, we recommend you use either [--follow-imports=normal](https://mypy.readthedocs.io/en/stable/command_line.html#cmdoption-mypy-follow-imports) (the default) or [--follow-imports=error](https://mypy.readthedocs.io/en/stable/command_line.html#cmdoption-mypy-follow-imports). Either option will help make sure you are not skipping checking any part of your codebase by accident.

> If you are planning on adding type hints to a large, existing code base, we recommend you start by trying to make your entire codebase (including files that do not use type hints) pass under [--follow-imports=normal](https://mypy.readthedocs.io/en/stable/command_line.html#cmdoption-mypy-follow-imports). This is usually not too difficult to do: mypy is designed to report as few error messages as possible when it is looking at unannotated code.

> **Only if doing this is intractable, we recommend passing mypy just the files you want to type check and use [--follow-imports=silent](https://mypy.readthedocs.io/en/stable/command_line.html#cmdoption-mypy-follow-imports).** Even if mypy is unable to perfectly type check a file, it can still glean some useful information by parsing it (for example, understanding what methods a given object has). See [Using mypy with an existing codebase](https://mypy.readthedocs.io/en/stable/existing_code.html#existing-code) for more recommendations.

Reply:
I don't think we should change it with the argument that what we want to check in the pipeline is everything as that is the end goal for the Panel project to have no issues whether internal facing or external facing.

I believe for "the current effort" which is focused on outward facing api we should add the relevant settings via flags manually when running mypy manually. Makes sense?